xchange-poloniex/src/main/java/org/knowm/xchange/poloniex/PoloniexAdapters.java

```diff
@@ -313,4 +316,59 @@ public static ExchangeMetaData adaptToExchangeMetaData(
     }
     return fundingRecords;
   }
+
+  public static Order adaptUserTradesToOrderStatus(PoloniexUserTrade[] poloniexUserTrades) {
+
+    if(poloniexUserTrades.length == 0)
+      return null;
+
+    Date date = null;
+    String id = null;
+    OrderType orderType = null;
+    CurrencyPair currencyPair = null;
+    BigDecimal amount = new BigDecimal(0);
+    BigDecimal fee = new BigDecimal(0);
+
+    List<BigDecimal> prices = new ArrayList<>();
+
+
+    //"2018-06-12 19:07:01"
+    SimpleDateFormat format = new SimpleDateFormat("yyyy-MM-dd hh:mm:ss");
+
+    for (PoloniexUserTrade poloniexUserTrade : poloniexUserTrades) {
+      id = poloniexUserTrade.getTradeID();
+      orderType = poloniexUserTrade.getType().equals("buy") ? OrderType.BID : OrderType.ASK; // what about others?
+
+      try {
+        date = format.parse(poloniexUserTrade.getDate());
+      } catch (ParseException e) {
+
+      }
+      amount = amount.add(poloniexUserTrade.getAmount());
+      fee = fee.add(poloniexUserTrade.getFee());
+
+      prices.add(poloniexUserTrade.getRate());
+
+    }
+
+
+
+    BigDecimal averagePrice = prices.stream().reduce(new BigDecimal(0), (a,b) -> a.add(b)).divide(new BigDecimal(poloniexUserTrades.length));
```

Comment:
this is wrong, you need to take each trades amount into consideration to calculate the proper average